app/controllers/katello/api/v2/debs_controller.rb

```diff
@@ -2,16 +2,69 @@ module Katello
   class Api::V2::DebsController < Api::V2::ApiController
     resource_description do
       name 'Deb Packages'
+      resource_id 'debs'
     end
-    apipie_concern_subst(:a_resource => N_("a deb"), :resource => "debs")
+    apipie_concern_subst(:a_resource => N_("a deb package"), :resource => "deb packages")
     include Katello::Concerns::Api::V2::RepositoryContentController
 
+    before_action :find_hosts, :only => :index
+
+    api :GET, "/debs", N_("List deb packages")
+    api :GET, "/content_views/:content_view_id/filters/:filter_id/debs", N_("List deb packages")
+    api :GET, "/content_view_filters/:content_view_filter_id/debs", N_("List deb packages")
+    api :GET, "/repositories/:repository_id/debs", N_("List deb packages")
+    param :organization_id, :number, :desc => N_("Organization identifier")
+    param :content_view_version_id, :number, :desc => N_("Content View Version identifier")
+    param :content_view_filter_id, :number, :desc => N_("Content View Filter identifier")
+    param :repository_id, :number, :desc => N_("Repository identifier")
+    param :environment_id, :number, :desc => N_("Environment identifier")
+    param :ids, Array, :desc => N_("Deb package identifiers to filter content by")
+    param :host_id, :number, :desc => N_("Host id to list applicable deb packages for")
+    param :packages_restrict_applicable, :boolean, :desc => N_("Return deb packages that are applicable to one or more hosts (defaults to true if host_id is specified)")
+    param :packages_restrict_upgradable, :boolean, :desc => N_("Return deb packages that are upgradable on one or more hosts")
+    #param :packages_restrict_latest, :boolean, :desc => N_("Return only the latest version of each package")
```

Comment:
Is this param needed?

Reply:
That one is available for RPM-packages, but I was not sure, whether it is applicable for DEB-packages as well.
IIRC deb-repos (in pulp) only publish the latest version of a package :thinking: 
@quba42 please correct me if I am wrong.

I have not really looked into implementing this, if it is possible and a valid use-case we should add it in a subsequent PR.
If not, I should drop the line :grin:

Reply:
I removed it for now. It is possible to implement it, but due to the limited possibility of comparing debian-versions I would like to have a separate PR to do this, because it makes sense to discuss performance a little bit for it.
I created a separate issue for it: https://projects.theforeman.org/issues/33525